tests/unit/sharded_test.cc

```diff
@@ -201,3 +201,32 @@ SEASTAR_THREAD_TEST_CASE(invoke_on_all_sharded_arg) {
     srv.stop().get();
     arg.stop().get();
 }
+
+SEASTAR_THREAD_TEST_CASE(invoke_on_modifiers) {
+    class checker {
+    public:
+        future<> fn(int a) {
+            return make_ready_future<>();
+        }
+        future<> fn(int a) const {
+            return make_ready_future<>();
+        }
+    };
+
+    seastar::sharded<checker> srv;
+    srv.start().get();
+    int a = 42;
+
+    srv.invoke_on_all([a] (checker& s) { return s.fn(a); }).get();
+    srv.invoke_on_all([a] (checker& s) mutable { return s.fn(a); }).get();
+    srv.invoke_on_others([a] (checker& s) { return s.fn(a); }).get();
+    srv.invoke_on_others([a] (checker& s) mutable ->future<> { return s.fn(a); }).get();
```

Comment:
how is this test case related to the fix?

Reply:
The lambda is marked `mutable`, this is what the fix fixes -- if you remove first patch from this PR and try to compile _this very line_ will fail compilation

Reply:
i meant, what's the point of `->future<>`?
how is 
```c++
srv.invoke_on_all([a] (checker& s) mutable { return s.fn(a); }).get();
```
different from
```
srv.invoke_on_all([a] (checker& s) mutable ->future<> { return s.fn(a); }).get();
```
?

Reply:
Ah, the `-> future<>` can be omitted indeed